src/v/redpanda/admin/server.cc

```diff
@@ -1577,12 +1576,28 @@ void config_multi_property_validation(
                 }
             }
         } break;
+        case model::cloud_credentials_source::azure_aks_oidc_federation: {
+            // for azure_aks_oidc_federation it is expected to receive part of
+            // the configuration via env variables. this check is just for
+            // related cluster properties
+            config_properties_seq properties = {
+              std::ref(updated_config.cloud_storage_azure_storage_account),
+              std::ref(updated_config.cloud_storage_azure_container),
+            };
+
+            for (auto& p : properties) {
+                if (p() == std::nullopt) {
+                    errors[ss::sstring(p.get().name())]
+                      = "Must be set when cloud storage enabled";
```

Comment:
Something like this will be more user friendly:

Similar for azure_vm_instance_metadata case below

```suggestion
                      = "Must be set when cloud storage enabled with credential source set to ...";
```

Reply:
can do it